hexrd/ui/image_stack_dialog.py

```diff
@@ -437,3 +439,20 @@ def show(self):
     def close_widget(self):
         if self.ui.isFloating():
             self.ui.close()
+
+    def detectors_changed(self):
+        self.detectors = HexrdConfig().detector_names
+        self.detector = self.detectors[0]
+        # update the state
+        self.state['dets'] = self.detectors
+        for det in self.detectors:
+            self.state[det] = {
```

Comment:
On a note separate to this PR: I see that we are setting the detector keys directly on `self.state` along with other keys [here](https://github.com/HEXRD/hexrdgui/blob/70fe5531afc768f3388a659ab5cd4df409e6db36/hexrd/ui/image_stack_dialog.py#L102-L116). If any detector names (which are user chosen) were to ever clash with these other names on `self.state`, it would cause a bug that would be hard to track down. So I'd recommend nesting the detector keys inside something else, like `self.state['det_settings'][det_name]` to avoid the possible clash. But I know the clash is unlikely and is therefore not urgent and doesn't need to be fixed on this PR.

Reply:
That's a fair point. Not for this PR like you mentioned, but this would be a good change. Along with that it would probably be smart to then move the detector settings (and others) into properties to make things simpler and cleaner.